### PR TITLE
integration/setup: fix tenant cluster deletion; improve logs

### PR DESCRIPTION
**Pull request by kopiczko**

### CODE REVIEW

**Comment by kopiczko** [comment]
Tests fail now because it takes time to delete the tenant cluster. 

Example: https://circleci.com/gh/giantswarm/aws-operator/14554

You can see it started to wait at `08:28:00` and timed out after ~1h at`09:29:07`.

There is no `event=delete` log in `aws-operator`.

Does that ring any bell?

```
{"caller":"github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:110","level":"debug","message":"waiting for deletion of CR `ci-wip-c1ac9-5e958` in namespace `default`","time":"2018-11-14T08:28:00.265378+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '523.233485ms' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:28:00.268604+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '1.110975214s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:28:00.795166+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '1.448612554s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:28:01.909398+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '2.439813377s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:28:03.361062+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '3.272951807s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:28:05.804477+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '5.501506204s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:28:09.080786+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '5.813913127s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:28:14.585713+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '8.297021018s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:28:20.403312+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '14.501103988s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:28:28.703201+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '17.963290195s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:28:43.209955+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '40.062066445s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:29:01.177253+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '40.188984277s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:29:41.243569+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '39.270041093s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:30:21.439791+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '2m4.305915725s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:31:00.713767+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '2m27.159314887s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:33:05.022973+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '3m3.241354489s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:35:32.185862+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '6m7.632989749s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:38:35.43426+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '3m39.067413068s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:44:43.070513+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '6m55.122387778s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:48:22.14114+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '5m47.284180932s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T08:55:17.267055+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '3m22.012338103s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T09:01:04.554605+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '3m26.256136669s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T09:04:26.570005+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '6m48.880122414s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T09:07:52.82942+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '4m23.723141144s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T09:14:41.712896+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '6m54.233440678s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T09:19:05.439021+00:00"}
{"caller":"github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '3m8.115724328s' due to error","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T09:25:59.675891+00:00"}
{"caller":"github.com/giantswarm/aws-operator/integration/setup/setup.go:51","level":"error","message":"failed to delete tenant cluster","stack":"[{/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:65: } {/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/e2e-harness/pkg/release/release.go:178: } {/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/e2e-harness/pkg/release/release.go:471: } {/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/vendor/github.com/giantswarm/e2e-harness/pkg/release/release.go:461: } {/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:126: } {/home/circleci/.go_workspace/src/github.com/giantswarm/aws-operator/integration/setup/tenant_cluster.go:120: CR `ci-wip-c1ac9-5e958` in namespace `default`} {still exists error}]","time":"2018-11-14T09:29:07.795401+00:00"}
```

File: integration/setup/tenant_cluster.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ func EnsureTenantClusterCreated(ctx context.Context, id string, config Config) e
 }
 
 func EnsureTenantClusterDeleted(ctx context.Context, id string, config Config) error {
-	err := config.Release.EnsureDeleted(ctx, id, CRNotExistsCondition(ctx, id, config))
+	err := config.Release.EnsureDeleted(ctx, awsConfigReleaseName(id), CRNotExistsCondition(ctx, id, config))
```
Comment (kopiczko) [inline integration/setup/tenant_cluster.go]: I think this line was making tests fail.

File: integration/setup/key.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// +build k8srequired
+
+package setup
+
+import (
+	"fmt"
+)
+
+func awsConfigReleaseName(clusterID string) string {
+	return fmt.Sprintf("e2esetup-awsconfig-%s", clusterID)
```
Comment (TheoBrigitte) [inline integration/setup/key.go]: Why do we need two packages with the exact same method and values ?
I am referring to `integration/key/release_name.go`

Comment (kopiczko) [reply integration/setup/key.go]: Because I left it half way during the transition :) Thanks!

**Comment by kopiczko** [comment]
Tests look good to me in the other PR based on this one with is here https://circleci.com/workflow-run/ea553b93-9090-489f-bb61-aef35f056ad6 

Two tests there failed because of `apprclient` timeout.
Other two failed because of `worker nodes are still not ready`.

Going.

**Pull request merged**